Client/Frontend/Home/RemoteTabsPanel.swift

```diff
@@ -82,28 +84,46 @@ class RemoteTabsPanel: UITableViewController, HomePanel {
             return
         }
 
-        // Otherwise, fetch the tabs cloud
-        self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
-            self.refreshControl?.endRefreshing()
-
+        self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
             if let clientAndTabs = result.successValue {
-                if clientAndTabs.count == 0 {
-                    self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .NoClients)
-                } else {
-                    let nonEmptyClientAndTabs = clientAndTabs.filter { $0.tabs.count > 0 }
-                    if nonEmptyClientAndTabs.count == 0 {
-                        self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .NoTabs)
-                    } else {
-                        self.tableViewDelegate = RemoteTabsPanelClientAndTabsDataSource(homePanel: self, clientAndTabs: nonEmptyClientAndTabs)
-                        self.tableView.allowsSelection = true
+                self.updateDelegateClientAndTabData(clientAndTabs)
+            } else {
+                self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .FailedToSync)
```

Comment:
Ya I don't like that either. I've moved this down below in the last else statement. So the logic path would be, grab the cached values, if that fails don't do anything until we know we can't sync. If we can't sync and the cached fetch didn't work, fail.